framework/src/variables/MooseVariableFE.C

```diff
@@ -1409,7 +1409,7 @@ MooseVariableFE<OutputType>::computeAD(
           _ad_second_u[qp] += _ad_dof_values[i] * (*second_phi)[i][qp];
       }
 
-      if (is_transient)
+      if (is_transient && _var_kind == Moose::VAR_NONLINEAR)
```

Comment:
I don't think is the right fix per se. There must be something stupid happening in the `TimeIntegrator::computeADTimeDerivatives` method which I'm going to look into now